---
string vendor_id
string product_id
string manufacturer
string product
